File: frontend/lib/models/house.dart
class House {
  final String id;
  final String address;
  final double latitude;
  final double longitude;
  final String riskLevel; // LOW, MODERATE, HIGH

  House({
    required this.id,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.riskLevel,
  });

  factory House.fromJson(Map<String, dynamic> json) {
    return House(
      id: json['_id'] ?? json['id'] ?? '',
      address: json['address'] ?? 'Unknown Address',
      latitude: (json['latitude'] ?? 0.0).toDouble(),
      longitude: (json['longitude'] ?? 0.0).toDouble(),
      riskLevel: json['riskLevel'] ?? 'LOW',
    );
  }
}

File: backend/models/House.js
const mongoose = require('mongoose');

const houseSchema = new mongoose.Schema(
  {
    address: {
      type: String,
      required: [true, 'Address is required'],
      trim: true,
    },
    latitude: {
      type: Number,
      required: [true, 'Latitude is required'],
    },
    longitude: {
      type: Number,
      required: [true, 'Longitude is required'],
    },
    riskLevel: {
      type: String,
      enum: ['LOW', 'MODERATE', 'HIGH'],
      default: 'LOW',
    },
    assignedStudentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: null,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('House', houseSchema);
